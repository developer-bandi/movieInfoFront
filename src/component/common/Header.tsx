import { Link } from 'react-router-dom';
import styled from 'styled-components';

interface HeaderProps {
  nick?: string;
  logout?: () => void;
}

const Header = ({ nick, logout }: HeaderProps) => {
  return (
    <HeaderBlock>
      <Title>오늘의 영화</Title>
      <UserInfoBlock>
        {nick ? (
          <>
            <LoginUser>{nick}</LoginUser>님 반갑습니다
            <LoginoutButton onClick={logout}>로그아웃</LoginoutButton>
          </>
        ) : (
          <LoginoutButton>
            <Link to={'/signin'}>로그인</Link>
          </LoginoutButton>
        )}
      </UserInfoBlock>
    </HeaderBlock>
  );
};

const HeaderBlock = styled.header`
  height: 5rem;
  border-bottom: 2px solid #eeeeee;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const Title = styled.h1`
  margin-left: 1rem;
  @media screen and (max-width: 500px) {
    font-size: 25px;
  }
`;

const UserInfoBlock = styled.div`
  margin-right: 1rem;
  display: flex;
  align-items: center;
`;

const LoginoutButton = styled.button`
  background: white;
  border: none;
  border: 1px solid black;
  border-radius: 3px;
  padding: 5px;
  margin-left: 1rem;
  cursor: pointer;
  @media screen and (max-width: 500px) {
    font-size: 10px;
  }
`;

const LoginUser = styled.div`
  margin-right: 5px;
  font-size: 20px;
  font-weight: bold;
  @media screen and (max-width: 500px) {
    font-size: 15px;
  }
`;

export default Header;
