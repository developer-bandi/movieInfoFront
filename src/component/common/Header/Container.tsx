import { useDispatch, useSelector } from 'react-redux';
import { ReducerType } from '../../../store';
import { doLogout } from '../../../store/user/Reducer';
import HeaderPresentational from './Presentational';

const HeaderContainer = () => {
  const userInfo = useSelector((state: ReducerType) => state.user);
  const dispatch = useDispatch();
  const logout = () => {
    dispatch(doLogout());
  };

  return <HeaderPresentational userInfo={userInfo} logout={logout} />;
};

export default HeaderContainer;
