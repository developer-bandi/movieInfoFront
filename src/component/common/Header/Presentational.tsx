import { Link } from 'react-router-dom';
import { UserState } from '../../../store/user/Reducer';
import styles from './Style';

interface HeaderProps {
  userInfo: UserState;
  logout: () => void;
}

const Header = ({ userInfo, logout }: HeaderProps) => {
  return (
    <styles.MainBlock>
      <styles.Title>오늘의 영화</styles.Title>
      <styles.UserInfoBlock>
        {userInfo.content !== undefined ? (
          <>
            <styles.Nickname>{userInfo.content?.nick}</styles.Nickname>님
            반갑습니다
            <styles.Button onClick={logout}>로그아웃</styles.Button>
          </>
        ) : (
          <styles.Button>
            <Link to={'/signin'}>로그인</Link>
          </styles.Button>
        )}
      </styles.UserInfoBlock>
    </styles.MainBlock>
  );
};

export default Header;
