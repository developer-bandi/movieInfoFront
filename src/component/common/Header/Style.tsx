import styled from 'styled-components';

const MainBlock = styled.header`
  height: 5rem;
  border-bottom: 2px solid #eeeeee;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const Title = styled.h1`
  margin-left: 1rem;
  @media screen and (max-width: 500px) {
    font-size: 25px;
  }
`;

const UserInfoBlock = styled.div`
  margin-right: 1rem;
  display: flex;
  align-items: center;
`;

const Button = styled.button`
  background: white;
  border: none;
  border: 1px solid black;
  border-radius: 3px;
  padding: 5px;
  margin-left: 1rem;
  cursor: pointer;
  @media screen and (max-width: 500px) {
    font-size: 10px;
  }
`;

const Nickname = styled.div`
  margin-right: 5px;
  font-size: 20px;
  font-weight: bold;
  @media screen and (max-width: 500px) {
    font-size: 15px;
  }
`;

const styles = { MainBlock, Title, UserInfoBlock, Button, Nickname };

export default styles;
