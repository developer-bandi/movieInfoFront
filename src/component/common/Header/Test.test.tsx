import { fireEvent, render, screen } from '@testing-library/react';
import Header from './Presentational';
import { MemoryRouter } from 'react-router-dom';
/// <reference types="@types/jest" />;
describe('Header Presentational', () => {
  it('로그인 하지 않았을 경우', () => {
    const utils = render(
      <Header userInfo={{ loading: false, error: false }} logout={jest.fn()} />,
      { wrapper: MemoryRouter }
    );
    expect(utils.container).toMatchSnapshot();
  });
  it('로그인 했을경우', () => {
    const logoutMockFn = jest.fn();
    const utils = render(
      <Header
        userInfo={{
          content: { id: 1, userid: 'test', nick: 'testname' },
          loading: false,
          error: false,
        }}
        logout={logoutMockFn}
      />,
      { wrapper: MemoryRouter }
    );
    fireEvent.click(screen.getByText('로그아웃'));
    expect(logoutMockFn).toBeCalledTimes(1);
    expect(utils.container).toMatchSnapshot();
  });
});
