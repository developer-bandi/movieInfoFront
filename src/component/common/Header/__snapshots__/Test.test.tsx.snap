// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Header 로그아웃 1`] = `
<div>
  <header
    class="sc-bdvvtL bOcBzn"
  >
    <h1
      class="sc-gsDKAQ kzGgeN"
    >
      오늘의 영화
    </h1>
    <div
      class="sc-dkPtRN dngAOt"
    >
      <button
        class="sc-hKwDye Spowr"
      >
        <a
          href="/signin"
        >
          로그인
        </a>
      </button>
    </div>
  </header>
</div>
`;

exports[`<Header/> Header 로그인 1`] = `
<div>
  <header
    class="sc-bdvvtL bOcBzn"
  >
    <h1
      class="sc-gsDKAQ kzGgeN"
    >
      오늘의 영화
    </h1>
    <div
      class="sc-dkPtRN dngAOt"
    >
      <div
        class="sc-eCImPb ldrnGy"
      >
        testname
      </div>
      님 반갑습니다
      <button
        class="sc-hKwDye Spowr"
      >
        로그아웃
      </button>
    </div>
  </header>
</div>
`;

exports[`Header Presentational 로그인 하지 않았을 경우 1`] = `
<div>
  <header
    class="sc-bdvvtL bOcBzn"
  >
    <h1
      class="sc-gsDKAQ kzGgeN"
    >
      오늘의 영화
    </h1>
    <div
      class="sc-dkPtRN dngAOt"
    >
      <button
        class="sc-hKwDye Spowr"
      >
        <a
          href="/signin"
        >
          로그인
        </a>
      </button>
    </div>
  </header>
</div>
`;

exports[`Header Presentational 로그인 했을경우 1`] = `
<div>
  <header
    class="sc-bdvvtL bOcBzn"
  >
    <h1
      class="sc-gsDKAQ kzGgeN"
    >
      오늘의 영화
    </h1>
    <div
      class="sc-dkPtRN dngAOt"
    >
      <div
        class="sc-eCImPb ldrnGy"
      >
        testname
      </div>
      님 반갑습니다
      <button
        class="sc-hKwDye Spowr"
      >
        로그아웃
      </button>
    </div>
  </header>
</div>
`;
